File: Examples/Objective-C/signature/signature/ViewController.m
//
//  ViewController.m
//  signature
//
//  Created by Vignesh on 2/10/16.
//  Copyright © 2016 vigneshuvi. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    CGFloat borderWidth = 2.0f;
    
    self.signatureImageView.layer.borderColor = [UIColor grayColor].CGColor;
    self.signatureImageView.layer.borderWidth = borderWidth;
}

-(void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

//implementation of delegate method
- (void)processCompleted:(UIImage*)signImage
{
    _signatureImageView.image = signImage;
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    if ([segue.identifier hasPrefix:@"view_to_capture"]) {
        CaptureSignatureViewController *destination = segue.destinationViewController;
        destination.delegate = self;
    }
}

#pragma mark - Sample protocol delegate


@end

File: Examples/Swift/signature/signature/ViewController.swift
//
//  ViewController.swift
//  signature
//
//  Created by Vignesh on 2/10/16.
//  Copyright © 2016 vigneshuvi. All rights reserved.
//

import UIKit



class ViewController: UIViewController, CaptureSignatureViewDelegate {
    
    var secondController: CaptureSignatureViewController?
    @IBOutlet var signatureImageView: UIImageView!

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        let borderWidth:CGFloat = 2.0
        
        self.signatureImageView.layer.borderColor = UIColor.gray.cgColor;
        self.signatureImageView.layer.borderWidth = borderWidth;
    }
    
    // implementation of the protocol
    func processCompleted(_ signImage: UIImage) {
        signatureImageView.image = signImage;
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if segue.identifier == "view_to_capture" {
            secondController = segue.destination as? CaptureSignatureViewController
            secondController!.delegate = self;
        }
    }


}

